tests/h/views/admin_groups_test.py

```diff
@@ -76,6 +78,39 @@ def call_on_success(request, form, on_success, on_failure):
         expected_location = pyramid_request.route_url('admin_groups')
         assert response == matchers.redirect_302_to(expected_location)
 
+    @pytest.mark.parametrize('type_', [
+        'open',
+        'restricted',
+    ])
+    def test_post_creates_group_on_success(self, pyramid_request, group_svc, handle_form_submission, type_):
+        name = 'My new group'
+        creator = pyramid_request.user.username
+        description = 'Purpose of new group'
+        origins = ['https://example.com']
+
+        def call_on_success(request, form, on_success, on_failure):
+            return on_success({
+                'authority': pyramid_request.authority,
+                'creator': creator,
+                'description': description,
+                'group_type': type_,
+                'name': name,
+                'origins': origins,
+            })
+        handle_form_submission.side_effect = call_on_success
+        ctrl = GroupCreateController(pyramid_request)
+
+        ctrl.post()
+
+        if type_ == 'open':
+            create_method = group_svc.create_open_group
+        else:
+            create_method = group_svc.create_restricted_group
+        expected_userid = 'acct:{}@{}'.format(creator, pyramid_request.authority)
+
+        create_method.assert_called_with(name=name, userid=expected_userid, description=description,
+                                         origins=origins)
```

Comment:
Ugh, wow, does `handle_form_submission` really require tests to be written like this?

I think you can get rid of the if/else on `type_` by moving `group_svc.create_open_group` and `group_svc.create_restricted_group` into the parametrize. You'll have to put the strings "create_open_group" and "create_restricted_group" into the parametrize and then in the test do `create_method = getattr(group_svc, create_method_name)`

Reply:
I wonder if a couple more tests could be added also? For example test the redirect that it returns?

Reply:
> I think you can get rid of the if/else on type_ by moving group_svc.create_open_group and group_svc.create_restricted_group into the parametrize. 

I could do, but I'm not convinced that this has any benefit given there are just two cases. IMO better to wait until new requirements come along to guide future changes. For example - perhaps the logic for creating different group types might differ in future and actually we'll end up splitting this test in two.

> Ugh, wow, does handle_form_submission really require tests to be written like this?

Yes. We could put together a helper to make writing this kind of test easier. I've followed the existing pattern here for consistency.

> I wonder if a couple more tests could be added also? For example test the redirect that it returns?

This is already checked by an existing test above.